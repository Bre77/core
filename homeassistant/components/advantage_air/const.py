"""Constants used by Advantage Air integration."""
DOMAIN = "advantage_air"
ADVANTAGE_AIR_AIRCONS = "aircons"
ADVANTAGE_AIR_LIGHTS = "myLights"
ADVANTAGE_AIR_THINGS = "myThings"
ADVANTAGE_AIR_RETRY = 10
ADVANTAGE_AIR_STATE_OPEN = "open"
ADVANTAGE_AIR_STATE_CLOSE = "close"
ADVANTAGE_AIR_STATE_ON = "on"
ADVANTAGE_AIR_STATE_OFF = "off"
ADVANTAGE_AIR_SET_AIRCON = "async_set_aircon"
ADVANTAGE_AIR_SET_LIGHT = "async_set_light"
ADVANTAGE_AIR_SET_THING = "async_set_thing"
ADVANTAGE_AIR_COORDINATOR = "coordinator"
