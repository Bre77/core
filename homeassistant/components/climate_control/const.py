"""Constants used by Climate Control integration."""
DOMAIN = "climate_control"
CONF_CLIMATE_ENTITY = "climate_entity"
CONF_AREAS = "areas"
CONF_ZONES = "zones"
CONF_COVER_ENTITY = "cover_entity"
CONF_SENSOR_ENTITY = "sensor_entity"
