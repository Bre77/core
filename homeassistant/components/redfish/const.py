"""Constants used by Refish integration."""
DOMAIN = "redfish"
REDFISH_POWER_ON = "On"
REDFISH_POWER_OFF = "Off"
