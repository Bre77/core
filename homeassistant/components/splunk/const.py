HEC_TIME = "time"
HEC_HOST = "host"
HEC_SOURCE = "source"
HEC_SOURCETYPE = "sourcetype"
HEC_INDEX = "index"
HEC_FIELDS = "fields"
HEC_EVENT = "event"
