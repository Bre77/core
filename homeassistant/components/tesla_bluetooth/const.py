"""Constants for the Tesla Bluetooth integration."""

import logging

DOMAIN = "tesla_bluetooth"
PRIVATE_KEY_FILE = "config/tesla_key.pem"
LOGGER = logging.getLogger(__package__)
MANUFACTURER_ID = 76
SERVICE_UUIDS = ["00001122-0000-1000-8000-00805f9b34fb"]

# UUIDs
GENERIC_ACCESS_SERVICE_UUID = "00001800-0000-1000-8000-00805f9b34fb"
DEVICE_NAME_UUID = "00002a00-0000-1000-8000-00805f9b34fb"
SEND_UUID = "00000212-b2d1-43f0-9b88-960cebf8b91e"
RECV_UUID = "00000213-b2d1-43f0-9b88-960cebf8b91e"
VERSION_UUID = "00000214-b2d1-43f0-9b88-960cebf8b91e"
