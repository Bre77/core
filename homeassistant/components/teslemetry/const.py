"""Constants used by Teslemetry integration."""
from __future__ import annotations

from enum import IntEnum, StrEnum

DOMAIN = "teslemetry"

MODELS = {
    "model3": "Model 3",
    "modelx": "Model X",
    "modely": "Model Y",
    "models": "Model S",
}


class TeslemetryStatus(StrEnum):
    """Teslemetry status."""

    ASLEEP = "asleep"
    ONLINE = "online"


class TeslemetrySeatHeaterOptions(StrEnum):
    """Teslemetry seat heater options."""

    OFF = "off"
    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"


class TeslemetryClimateKeeper(StrEnum):
    """Teslemetry Climate Keeper Modes."""

    OFF = "off"
    ON = "on"
    DOG = "dog"
    CAMP = "camp"


class TeslemetryUpdateStatus(StrEnum):
    """Teslemetry Update Statuses."""

    AVAILABLE = "available"
    DOWNLOADING = "downloading"
    INSTALLING = "installing"
    WIFI_WAIT = "downloading_wifi_wait"
    SCHEDULED = "scheduled"


class TeslemetryCoverStates(IntEnum):
    """Teslemetry Cover states."""

    CLOSED = 0
    OPEN = 1
