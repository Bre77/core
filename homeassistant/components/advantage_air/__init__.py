"""Advantage Air climate integration."""
from datetime import timedelta
import logging

from advantage_air import ApiError, advantage_air

from homeassistant.config_entries import ConfigEntry
from homeassistant.const import CONF_IP_ADDRESS, CONF_PORT, Platform
from homeassistant.core import HomeAssistant
from homeassistant.helpers.aiohttp_client import async_get_clientsession
from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed

from .const import (
    ADVANTAGE_AIR_COORDINATOR,
    ADVANTAGE_AIR_RETRY,
    ADVANTAGE_AIR_SET_AIRCON,
    ADVANTAGE_AIR_SET_LIGHT,
    DOMAIN,
)

ADVANTAGE_AIR_SYNC_INTERVAL = 15
PLATFORMS = [
    Platform.BINARY_SENSOR,
    Platform.CLIMATE,
    Platform.COVER,
    Platform.SELECT,
    Platform.SENSOR,
    Platform.SWITCH,
    Platform.LIGHT,
]

_LOGGER = logging.getLogger(__name__)


async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
    """Set up Advantage Air config."""
    ip_address = entry.data[CONF_IP_ADDRESS]
    port = entry.data[CONF_PORT]
    api = advantage_air(
        ip_address,
        port=port,
        session=async_get_clientsession(hass),
        retry=ADVANTAGE_AIR_RETRY,
    )

    async def async_get():
        try:
            return await api.async_get()
        except ApiError as err:
            raise UpdateFailed(err) from err

    coordinator = DataUpdateCoordinator(
        hass,
        _LOGGER,
        name="Advantage Air",
        update_method=async_get,
        update_interval=timedelta(seconds=ADVANTAGE_AIR_SYNC_INTERVAL),
    )

    async def async_set_aircon(change):
        try:
            if await api.aircon.async_set(change):
                await coordinator.async_refresh()
        except ApiError as err:
            _LOGGER.warning(err)

    async def async_set_light(change):
        try:
            if await api.lights.async_set(change):
                await coordinator.async_refresh()
        except ApiError as err:
            _LOGGER.warning(err)

    await coordinator.async_config_entry_first_refresh()

    hass.data.setdefault(DOMAIN, {})
    hass.data[DOMAIN][entry.entry_id] = {
        ADVANTAGE_AIR_COORDINATOR: coordinator,
        ADVANTAGE_AIR_SET_AIRCON: async_set_aircon,
        ADVANTAGE_AIR_SET_LIGHT: async_set_light,
    }

    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)

    return True


async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
    """Unload Advantage Air Config."""
    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)

    if unload_ok:
        hass.data[DOMAIN].pop(entry.entry_id)

    return unload_ok
