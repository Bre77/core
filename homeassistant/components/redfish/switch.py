"""Switch platform for Redfish integration."""

from homeassistant.helpers.entity import ToggleEntity

from .const import DOMAIN as REDFISH_DOMAIN, REDFISH_POWER_ON
from .entity import RedfishEntity

# REDFISH_RESETTYPE_ON = "On"
# REDFISH_RESETTYPE_OFF = "On"


async def async_setup_entry(hass, config_entry, async_add_entities):
    """Set up Redfish toggle platform."""

    instance = hass.data[REDFISH_DOMAIN][config_entry.entry_id]

    entities = []
    for id in instance["coordinator"].data:
        entities.append(RedfishPower(instance, id))
    async_add_entities(entities)


class RedfishPower(RedfishEntity, ToggleEntity):
    """Representation of Redfish fresh air control."""

    @property
    def name(self):
        """Return the name."""
        return f"{self._name} Power"

    @property
    def unique_id(self):
        """Return a unique id."""
        return f"{self._uid}-power"

    @property
    def is_on(self):
        """Return the power state."""
        return self._data["Systems"]["PowerState"] == REDFISH_POWER_ON

    @property
    def icon(self):
        """Return a representative icon of the power switch."""
        return ["mdi:server-network-off", "mdi:server-network"][self.is_on]

    async def async_turn_on(self, **kwargs):
        """Turn power on."""
        if not self.is_on:
            await self.async_change(
                self._data["Systems"]["Actions"]["#ComputerSystem.Reset"]["target"],
                {"ResetType": "PushPowerButton"},
            )

    async def async_turn_off(self, **kwargs):
        """Turn power off."""
        if self.is_on:
            await self.async_change(
                self._data["Systems"]["Actions"]["#ComputerSystem.Reset"]["target"],
                {"ResetType": "PushPowerButton"},
            )
