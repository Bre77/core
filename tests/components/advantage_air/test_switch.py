"""Test the Advantage Air Switch Platform."""
from json import loads

from homeassistant.components.advantage_air.const import (
    ADVANTAGE_AIR_STATE_OFF,
    ADVANTAGE_AIR_STATE_ON,
)
from homeassistant.components.switch import (
    DOMAIN as SWITCH_DOMAIN,
    SERVICE_TURN_OFF,
    SERVICE_TURN_ON,
)
from homeassistant.const import ATTR_ENTITY_ID, STATE_OFF, STATE_ON
from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er

from . import (
    TEST_SET_RESPONSE,
    TEST_SET_THING_URL,
    TEST_SET_URL,
    TEST_SYSTEM_DATA,
    TEST_SYSTEM_URL,
    add_mock_config,
)

from tests.test_util.aiohttp import AiohttpClientMocker


async def test_cover_async_setup_entry(
    hass: HomeAssistant,
    aioclient_mock: AiohttpClientMocker,
    entity_registry: er.EntityRegistry,
) -> None:
    """Test switch platform."""

    aioclient_mock.get(
        TEST_SYSTEM_URL,
        text=TEST_SYSTEM_DATA,
    )
    aioclient_mock.get(
        TEST_SET_URL,
        text=TEST_SET_RESPONSE,
    )

    await add_mock_config(hass)

    # Test Switch Entity
    entity_id = "switch.myzone_fresh_air"
    state = hass.states.get(entity_id)
    assert state
    assert state.state == STATE_OFF

    entry = entity_registry.async_get(entity_id)
    assert entry
    assert entry.unique_id == "uniqueid-ac1-freshair"

    await hass.services.async_call(
        SWITCH_DOMAIN,
        SERVICE_TURN_ON,
        {ATTR_ENTITY_ID: [entity_id]},
        blocking=True,
    )

    await hass.services.async_call(
        SWITCH_DOMAIN,
        SERVICE_TURN_OFF,
        {ATTR_ENTITY_ID: [entity_id]},
        blocking=True,
    )


async def test_things_switch(
    hass: HomeAssistant,
    aioclient_mock: AiohttpClientMocker,
    entity_registry: er.EntityRegistry,
) -> None:
    """Test things switches."""

    aioclient_mock.get(
        TEST_SYSTEM_URL,
        text=TEST_SYSTEM_DATA,
    )
    aioclient_mock.get(
        TEST_SET_THING_URL,
        text=TEST_SET_RESPONSE,
    )

    await add_mock_config(hass)

    # Test Switch Entity
    entity_id = "switch.relay"
    thing_id = "205"
    state = hass.states.get(entity_id)
    assert state
    assert state.state == STATE_ON

    entry = entity_registry.async_get(entity_id)
    assert entry
    assert entry.unique_id == "uniqueid-205"

    await hass.services.async_call(
        SWITCH_DOMAIN,
        SERVICE_TURN_OFF,
        {ATTR_ENTITY_ID: [entity_id]},
        blocking=True,
    )

    await hass.services.async_call(
        SWITCH_DOMAIN,
        SERVICE_TURN_ON,
        {ATTR_ENTITY_ID: [entity_id]},
        blocking=True,
    )
