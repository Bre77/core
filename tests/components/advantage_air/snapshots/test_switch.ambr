# serializer version: 1
# name: test_cover_async_setup_entry[switch.myzone_myfan-turnoff]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'switch',
      'friendly_name': 'myzone MyFan',
      'icon': 'mdi:fan-auto',
    }),
    'context': <ANY>,
    'entity_id': 'switch.myzone_myfan',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_cover_async_setup_entry[switch.myzone_myfan-turnon]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'switch',
      'friendly_name': 'myzone MyFan',
      'icon': 'mdi:fan-auto',
    }),
    'context': <ANY>,
    'entity_id': 'switch.myzone_myfan',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_cover_async_setup_entry[switch.myzone_myfan]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'switch',
      'friendly_name': 'myzone MyFan',
      'icon': 'mdi:fan-auto',
    }),
    'context': <ANY>,
    'entity_id': 'switch.myzone_myfan',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
